### DISCUSSION TITLE
Minimum Requirements

### DISCUSSION THREAD

**Discussion by ohadsamu** [grafana/mimir#6948] [Help and support]
Hi, we were looking for minimum requirements for capacity planning since we don't have the number of time series yet, and we want to deploy the smallest cluster for test porposes.
Is there any recommendation for minimum sizes of each microservice? Thanks in advance.

**Comment by pracucci** [comment]
Hi! Mimir resources utilization (CPU, memory, storage) is mostly based on the amount of series and samples ingested and queried. This means that if you have zero traffic, both on read and write path, you should expect a very low baseline utilisation.

That being said, Mimir is a distributed time series database designed and built to run at large scale. Most of the Mimir example configurations assumes at least some significant traffic. For example, the [`small.yaml`](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/small.yaml) scale config in [Mimir helm chart](https://grafana.com/docs/helm-charts/mimir-distributed/latest/) assumes 1 million active series on the write path (which is a small scale for Mimir, but probably larger than "the smallest cluster").

My suggestion is to deploy Mimir on K8S using the Helm chart, set resources based on `small.yaml` config and eventually override resources request based on actual utilization in your test cluster.

**Comment by ohadsamu** [comment]
Hi! Thank you very much for the information!
Have a nice day,
Ohad.
